frame/aura/src/lib.rs

```diff
@@ -126,18 +133,31 @@ pub mod pallet {
 }
 
 impl<T: Config> Pallet<T> {
+	fn to_bounded_vec(
+		authorities: Vec<T::AuthorityId>,
+	) -> BoundedVec<T::AuthorityId, T::MaxAuthorities> {
+		let bounded_authorities =
+			BoundedVec::<T::AuthorityId, T::MaxAuthorities>::try_from(authorities);
+		assert!(
+			bounded_authorities.is_ok(),
+			"More than the maximum number of authorities provided"
+		);
```

Comment:
you are not allowed to panic in the runtime. You need to change these things to return a result

Reply:
Or have the input itself already be bounded, but probably returning the result is better

Reply:
Amended the way I convert to BoundedVec, it's called in 2 locations:
initialise_keys, this one already panics, so no harm in panicking if size is too large.
on_new_session, this could be a contentious choice, I have truncated the provided vec to fit the max size, and then we know this cannot panic. When I do truncate, I log this to pass info to the user.